src/hotspot/share/services/memReporter.cpp

```diff
@@ -22,20 +22,22 @@
  *
  */
 #include "precompiled.hpp"
+#include "jfr/jfrEvents.hpp"
 #include "memory/allocation.hpp"
 #include "memory/metaspace.hpp"
 #include "memory/metaspaceUtils.hpp"
 #include "services/mallocTracker.hpp"
 #include "services/memReporter.hpp"
+#include "services/memTracker.hpp"
 #include "services/threadStackTracker.hpp"
 #include "services/virtualMemoryTracker.hpp"
 #include "utilities/globalDefinitions.hpp"
 
-size_t MemReporterBase::reserved_total(const MallocMemory* malloc, const VirtualMemory* vm) const {
+size_t MemReporterBase::reserved_total(const MallocMemory* malloc, const VirtualMemory* vm) {
```

Comment:
I turned `reserved_total(...)` and `committed_total(...)`into a static helpers since they did not use any members. That way my new reporter did not have to inherit `MemReporter`.

Reply:
Ah okay, I was blind, sorry.

Reply:
No problem, I actually asked my self the same question when I did a quick browse of the code before publishing :)